docs/overview/architecture/proposal-functions.md

```diff
@@ -0,0 +1,37 @@
+The [Proposal.sol](../../../src/proposals/Proposal.sol) file contains a set of functions that all governance models inherit. There are three governance models: Bravo, Multisig and Timelock already implemented in the example repository. To create a Proposal with FPS, you can inherit one of the governance types, such as [MultisigProposal.sol](../../../src/proposals/MultisigProposal.sol) and override the needed functions. Alternatively, you can create a new governance model in your repository that inherits the `Proposal` contract and then a proposal can be created using a newly formed governance model not already inside the repo. The following is a list of functions all proposals must implement:
```

Comment:
```suggestion
The [Proposal.sol](../../../src/proposals/Proposal.sol) file contains a set of functions that all governance models inherit. There are currently four governance models inheriting from Proposal.sol: Bravo, Multisig, Timelock and OpenZeppelin Governor. When using FPS for any of the abovementioned models for creating proposals, all that needs to be done is to inherit one of the proposal types, such as [GovernorBravoProposal.sol](../../../src/proposals/GovernorBravoProposal.sol) and override the necessary functions to create the proposal, like `build` and `deploy`. FPS is flexible enough so that for any different governance model, governance proposal types can be easily adjusted to fit into the governance architecture. An example has been provided using Arbitrum Governance on FPS example repo (link here) to demonstrate FPS flexibility. The following is a list of functions proposals can implement:
```